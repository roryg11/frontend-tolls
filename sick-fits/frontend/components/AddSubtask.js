import React, { Component } from 'react';
import { Mutation } from "react-apollo";
import gql from "graphql-tag";
import styled from "styled-components";
import Form from "./styles/Form";
import Error from "./ErrorMessage";
import SickButton from "./styles/SickButton";
import { GOAL_TASK_QUERY } from "./Task";

const CREATE_SUBTASK_MUTATION = gql`
    mutation CREATE_SUBTASK_MUTATION (
        $name: String!
        $description: String
        $taskId: ID!
    ) {
        createSubTask(
            name: $name
            description: $description
            taskId: $taskId
        ) {
            id
            name
            description
        }
    }
`

const FormStyles = styled.div`
    padding: 20px;
`

class AddSubtask extends Component {
    state = {
        taskId: this.props.taskId,
        name: "",
        description: ""
    }

    onChange = (e) => {
        const {name, value} = e.target; 
        this.setState({[name]: value});
    }
    render() {
        // will need to add refetch queries and animate the appearance of the subtask
        return (
            <Mutation mutation={CREATE_SUBTASK_MUTATION} variables={this.state} refetchQueries={[{query: GOAL_TASK_QUERY, variables: {id: this.props.taskId}}]}>
                { (addSubtask, {loading, error}) => {
                    if(loading) return <p>Loading...</p>;
                    if(error) return <Error error={error}/>;
                    return (
                        <FormStyles>
                            <Form onSubmit={
                            (e)=>{
                                e.preventDefault();
                                addSubtask();
                            }
                            }>
                                <label htmlFor="name">
                                        Name
                                        <input type="text"
                                        id="name"
                                        name="name"
                                        placeholder="Name"
                                        required
                                        value={this.state.name}
                                        onChange={this.onChange}/>
                                    </label>
                                    <label htmlFor="description">
                                        Description
                                        <input type="text"
                                        id="description"
                                        name="description"
                                        placeholder="How does this help achieve your milestone?"
                                        required
                                        value={this.state.description}
                                        onChange={this.onChange}/>
                                    </label>
                                    <SickButton type="Submit">Add Subtask</SickButton>
                            </Form>
                        </FormStyles>
                        
                    )
                }}
            </Mutation>
        );
    }
}

export default AddSubtask;