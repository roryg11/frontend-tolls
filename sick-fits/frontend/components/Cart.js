import {Query, Mutation} from "react-apollo";
import gql from "graphql-tag";
import CartStyles from "./styles/CartStyles"
import CloseButton from "./styles/CloseButton"
import SickButton from "./styles/SickButton"
import Supreme from "./styles/Supreme"

const LOCAL_STATE_QUERY = gql`
    query {
        cartOpen @client
    }
`

const TOGGLE_CART_MUTATION = gql`
    mutation {
        toggleCart @client
    }
`

const Cart = props => (
    <Mutation mutation={TOGGLE_CART_MUTATION}>
        { (toggleCart)=>(
            <Query query={LOCAL_STATE_QUERY}>
                { ({data}) => (
                    <CartStyles open={data.cartOpen}>
                        <header>
                            <CloseButton title="close" onClick={toggleCart}>&times;</CloseButton>
                            <Supreme>Your Cart</Supreme>
                            <p>You have __ items in your cart</p>
                        </header>
                        <footer>
                            <p>$10.10</p>
                            <SickButton>Checkout</SickButton>
                        </footer>
                    </CartStyles>
                )}
            </Query>
        )}
    </Mutation>
)

export default Cart; 
export { LOCAL_STATE_QUERY };
export { TOGGLE_CART_MUTATION }; 