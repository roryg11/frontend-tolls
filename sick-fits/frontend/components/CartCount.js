import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { TransitionGroup, CSSTransition} from "react-transition-group";

const AnimationStyles = styled.span`
    position: relative;
    .count {
        display: block;
        position: relative;
        transition: all 0.4s;
        backface-visibility: hidden;
    }
    /* initial state of entered dot flipped on back */
    /* one turn is 360 degrees */
    .count-enter{
        transform: rotateX(0.5turn);
    }
    .count-enter-active {
        transform: rotateX(0);
    }

    .count-exit {
        top: 0;
        position: absolute;
        transform: rotateX(0);
    }

    .count-exit-active {
        transform: rotateX(0.5turn);
    }
`;


// font-feature-settings and font-variant-numeric help wiht
// the fact that numbers have different widths -- this maintains the same width
const Dot = styled.div`
    background-color: ${props => props.theme.red};
    color: white;
    border-radius: 50%;
    padding: 0.5rem;
    line-height: 2rem;
    width: 3rem;
    margin-left: 1rem;
    font-weight: 100;
    font-feature-settings: tnum;
    font-variant-numeric: tabular-nums;
`

const CartCount = ({count}) => (
    <AnimationStyles>
        <TransitionGroup>
            <CSSTransition 
                unmountOnExit 
                className="count" 
                classNames="count" 
                key={count} 
                timeout={{enter: 400, exit: 400}}>
                <Dot>{count}</Dot>
            </CSSTransition>
        </TransitionGroup>
    </AnimationStyles>
    
)

export default CartCount;