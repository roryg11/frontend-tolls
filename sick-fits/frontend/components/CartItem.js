import styled from "styled-components";
import formatMoney from "../lib/formatMoney";
import RemoveFromCart from "./RemoveFromCart";

const CartStyles = styled.li`
    padding: 1rem 0;
    display: grid;
    grid-template-columns: auto 1fr auto;
    border-bottom: 1px solid ${ props => props.theme.lightGrey};
    align-items: center;
    img {
        margin-right: 10px;
    }
    h3, p {
        margin: 0; 
    }
`

const CartItem = props => {
    const {cartItem} = props;
    if(!cartItem.item){ 
        return (
            <CartStyles>
                <div>&nbsp;</div>
                <p>This item has been removed</p>
                <RemoveFromCart id={cartItem.id}/>
            </CartStyles>
        )
    };
    return (
        <CartStyles>
            <img width="100" src={cartItem.item.image}/>
            <div className="cart-item-details">
                <h3>{cartItem.item.title}</h3>
                <p>
                    {formatMoney(cartItem.item.price * cartItem.quantity)}
                    {' - '}
                    <em>
                        {cartItem.quantity} &times; {formatMoney(cartItem.item.price)} each
                    </em>
                </p>
            </div>
            <RemoveFromCart id={cartItem.id}/>
        </CartStyles>
    )
}

export default CartItem; 