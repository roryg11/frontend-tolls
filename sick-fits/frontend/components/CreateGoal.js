import React, { Component } from 'react';
import { Mutation } from "react-apollo";
import gql from "graphql-tag";
import Router from "next/router";
import styled from "styled-components";
import Error from "./ErrorMessage";
import Form from "./styles/Form";
import SickButton from "./styles/SickButton";

const MEASUREMENTS = [
    "FREQUENCY",
    "LIST"
]; 

const CREATE_GOAL_MUTATION = gql`
    mutation CREATE_GOAL_MUTATION(
        $name: String!
        $description: String!
        $measurement: Measurement
    ) {
        createGoal(
            name: $name,
            description: $description,
            measurement: $measurement
        ) {
            id
            name
        }
    }
`

// Future to-do, add inspiration images etc

class CreateGoal extends Component {
    state = {
        name: "",
        description: "",
        measurement: MEASUREMENTS.frequency
    }
    onChange = (e) => {
        const { name, value } = e.target; 
        this.setState({[name]: value}); 
    }
    render(){
        return(
            <Mutation mutation={CREATE_GOAL_MUTATION} variables={this.state}>
                { (createGoal, {loading, error}) => {
                        if(error) return <Error error={error}/>
                        if(loading) return <p>Loading...</p>
                        return (
                            <Form onSubmit={ async (e) => {
                                e.preventDefault();

                                const res = await createGoal();
                                Router.push({
                                    pathname: "/goal", 
                                    query: {id: res.data.createGoal.id}
                                });
                            }
                            }>
                                 <label htmlFor="name">
                                    Name
                                    <input type="text"
                                    id="name"
                                    name="name"
                                    placeholder="Name"
                                    required
                                    value={this.state.name}
                                    onChange={this.onChange}/>
                                </label>
                                <label htmlFor="description">
                                    Description
                                    <input type="text"
                                    id="description"
                                    name="description"
                                    placeholder="Why is this your goal?"
                                    required
                                    value={this.state.description}
                                    onChange={this.onChange}/>
                                </label>
                                <label htmlFor="measurement">
                                    Measurement - how will you measure your success?
                                    <select onChange={this.onChange} name="measurement" defaultValue={MEASUREMENTS.frequency}>
                                        {
                                            MEASUREMENTS.map((measurement)=>{
                                                return <option key={measurement} value={measurement}>{measurement}</option>;    
                                            })
                                        }
                                    </select>
                                </label>
                                <SickButton type="Submit">Onward!</SickButton>
                            </Form>
                        )
                    }
                }
            </Mutation>
        )
    }
}

export default CreateGoal;
export {MEASUREMENTS}; 

