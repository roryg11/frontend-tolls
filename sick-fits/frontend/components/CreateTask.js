import React, { Component } from 'react';
import { Mutation } from 'react-apollo';
import gql from "graphql-tag";
import Router from "next/router";
import styled from "styled-components";
import Form from "./styles/Form";

const CREATE_TASK_MUTATION = gql`
    mutation CREATE_TASK_MUTATION (
        $name: String!
        $description: String
        $goalId: ID!
    ) {
        createTask(
            name: $name
            description: $description
            goalId: $goalId
        ) {
            id
            name
            description
        }
    }
`


class CreateTask extends Component {
    state = {
        name: "",
        description: "",
        goalId: this.props.goalId
    }

    handleChange = (e) => {
        const { name, value } = e.target; 
        this.setState({[name]: value});
    }

    render() {
        return (
            <Mutation mutation={CREATE_TASK_MUTATION} variables={this.state}>
                { (createTask, {loading, error})=>{
                        return (<Form onSubmit={
                                        async e=> {
                                            // stop form from submitting
                                            e.preventDefault();
                                            //call the mutation
                                            const res = await createTask();
                                            // change to the single item page
                                            Router.push({
                                                pathname: "/milestone", 
                                                query: {id: res.data.createTask.id}
                                            });
                                        }
                            }>
                            <fieldset disabled={loading} aria-busy={loading}>
                                <label htmlFor="name">
                                    Name
                                    <input type="text"
                                    id="name"
                                    name="name"
                                    placeholder="name"
                                    required
                                    value={this.state.title}
                                    onChange={this.handleChange}/>
                                </label>
                                <label htmlFor="description">
                                    Description
                                    <textarea
                                    id="description"
                                    name="description"
                                    placeholder="Enter a description"
                                    required
                                    value={this.state.description}
                                    onChange={this.handleChange}/>
                                </label>
                                <button type="Submit">Submit</button>
                            </fieldset>
                        </Form>)
                    }
                }
            </Mutation>
        );
    }
}

export default CreateTask;