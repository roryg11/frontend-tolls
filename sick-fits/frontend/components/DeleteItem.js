import React, { Component } from 'react';
import {Mutation} from "react-apollo";
import gql from "graphql-tag";
import {ALL_ITEMS_QUERY} from "./Items";

const DELETE_ITEM_MUTATION = gql`
    mutation DELETE_ITEM_MUTATIAON($id: ID!){
        deleteItem(id: $id){
            id
        }
    }
`

class DeleteItem extends Component {
    update = (cache, payload) => {
        const data = cache.readQuery({query: ALL_ITEMS_QUERY});
        // manually update the cache to reflect the backend
        // filter deleted item out of the page
        data.items = data.items.filter(item=> item.id !== payload.data.deleteItem.id);
        cache.writeQuery({query: ALL_ITEMS_QUERY, data});
    }
    render() {
        return (
            <Mutation 
            mutation={DELETE_ITEM_MUTATION} 
            variables={{id: this.props.id}}
            update={this.update}>
                {
                    (deleteItem, {error})=>{
                        return <button onClick={()=>{
                            if(confirm('Are you sure you want to delete this item?')){
                                deleteItem().catch((error)=>{
                                    alert(error);
                                });
                            }
                        }}
                        >{this.props.children}</button>
                    }
                }
            </Mutation>
            
        );
    }
}

export default DeleteItem;