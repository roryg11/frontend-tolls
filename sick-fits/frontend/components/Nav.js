import Link from "next/link";
import { Mutation } from "react-apollo";
import NavStyles from "./styles/NavStyles";
import User from "./User";
import Signout from "./Signout";
import CartCount from "./CartCount";
import { TOGGLE_CART_MUTATION } from "../components/Cart";

const Nav = () => (
        <User>
            { ({data: {me}}) => (
                <NavStyles>
                    {me && (
                        <>
                            <Link href="/goals">
                                <a>Goals</a>
                            </Link>
                            <Link href="/addGoals">
                                <a>Add Goal</a>
                            </Link>
                            <Link href="/me">
                                <a>Account</a>
                            </Link>
                            <Signout/>
                            <Mutation mutation={TOGGLE_CART_MUTATION}>
                                { (toggleCart)=>(
                                    <button onClick={toggleCart}>My Cart
                                    <CartCount count={me.cart.reduce((tally, cartItem)=>{
                                        return tally + cartItem.quantity
                                    }, 0)}/>
                                    </button>
                                ) }
                            </Mutation>
                        </>
                    )}

                    {!me && (
                        <Link href="/signup">
                            <a>Signup</a>
                        </Link>
                    )}
                    <Link href="/goals">
                        <a>Home</a>
                    </Link>
                </NavStyles>
                )
            }
        </User>
        
)

export default Nav; 