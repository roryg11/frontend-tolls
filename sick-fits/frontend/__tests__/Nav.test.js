import {shallow, mount} from "enzyme";
import toJSON from "enzyme-to-json";
import { MockedProvider } from "react-apollo/test-utils";
import wait from "waait";
import Nav from "../components/Nav";
import { CURRENT_USER_QUERY } from "../components/user";
import { fakeUser, fakeCartItem } from "../lib/testUtils";

const notSignedIn = [
    {
        request: { query: CURRENT_USER_QUERY },
        result: { data: {me: null}}
    }
];

const signedInMock = [
    {
        request: {query: CURRENT_USER_QUERY },
        result: { data: {me: fakeUser()}}
    }
];

const signedInWithItems = [
    {
        request: {query: CURRENT_USER_QUERY},
        result: { data: { me: {
                ...fakeUser(),
                cart: [fakeCartItem(), fakeCartItem(), fakeCartItem()]
            }
        }
    }
    }
]

describe('Nav', ()=>{
    it('shows a simple nav when a user is not signed in', async ()=>{
        const wrapper = mount(
            <MockedProvider mocks={notSignedIn}>
                <Nav/>
            </MockedProvider>
        );
        
        await wait();
        wrapper.update();
        const links = wrapper.find('Link');
        const nav = wrapper.find('Nav');
        expect(links.length).toEqual(3);
        expect(toJSON(nav)).toMatchSnapshot();
;    });
    it('shows the full nav when a user is logged in', async ()=>{
        const wrapper = mount(
            <MockedProvider mocks={signedInMock}>
                <Nav/>
            </MockedProvider>
        );

        await wait();
        wrapper.update();
        const links = wrapper.find('Link');
        expect(links.length).toEqual(5);
        expect(wrapper.text()).toContain("Shop");
        expect(wrapper.text()).toContain("Sell");
        expect(wrapper.text()).toContain("Orders");
        expect(wrapper.text()).toContain("Account");
        expect(wrapper.find('Signout').exists()).toBe(true);
    });
    it('shows the cart count when the user has items in their cart', async ()=>{
        const wrapper = mount(
            <MockedProvider mocks={signedInWithItems}>
                <Nav/>
            </MockedProvider>
        );

        await wait();
        wrapper.update();
        const count = wrapper.find('.count');
        expect(toJSON(count)).toMatchSnapshot();
    });
});