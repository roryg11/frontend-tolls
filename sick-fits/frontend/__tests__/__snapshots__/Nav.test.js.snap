// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav shows a simple nav when a user is not signed in 1`] = `
<Nav>
  <User>
    <Query
      query={
        Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": undefined,
              "operation": "query",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "me",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "id",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "name",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "email",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "permissions",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "orders",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "cart",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "quantity",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "item",
                                },
                                "selectionSet": Object {
                                  "kind": "SelectionSet",
                                  "selections": Array [
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "id",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "title",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "description",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "image",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "price",
                                      },
                                      "selectionSet": undefined,
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 419,
            "start": 0,
          },
        }
      }
    >
      <NavStyles>
        <ul
          className="NavStyles-sc-11c0d2g-0 iTUtFG"
        >
          <Link
            href="/items"
          >
            <a
              href="/items"
              onClick={[Function]}
            >
              Shop
            </a>
          </Link>
          <Link
            href="/signup"
          >
            <a
              href="/signup"
              onClick={[Function]}
            >
              Signup
            </a>
          </Link>
          <Link
            href="/"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </Link>
        </ul>
      </NavStyles>
    </Query>
  </User>
</Nav>
`;

exports[`Nav shows the cart count when the user has items in their cart 1`] = `
Array [
  <CSSTransition
    className="count"
    classNames="count"
    in={true}
    key=".$9"
    onExited={[Function]}
    timeout={
      Object {
        "enter": 400,
        "exit": 400,
      }
    }
    unmountOnExit={true}
  >
    <Transition
      appear={false}
      className="count"
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot
        className="count"
      >
        <div
          className="count CartCount__Dot-xxvp4g-1 fKxapv"
        >
          9
        </div>
      </CartCount__Dot>
    </Transition>
  </CSSTransition>,
  <Transition
    appear={false}
    className="count"
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={
      Object {
        "enter": 400,
        "exit": 400,
      }
    }
    unmountOnExit={true}
  >
    <CartCount__Dot
      className="count"
    >
      <div
        className="count CartCount__Dot-xxvp4g-1 fKxapv"
      >
        9
      </div>
    </CartCount__Dot>
  </Transition>,
  <CartCount__Dot
    className="count"
  >
    <div
      className="count CartCount__Dot-xxvp4g-1 fKxapv"
    >
      9
    </div>
  </CartCount__Dot>,
  <div
    className="count CartCount__Dot-xxvp4g-1 fKxapv"
  >
    9
  </div>,
]
`;
